Wikipedia/Code/WMFScreenHeightConstrainedGradientTextView.swift

```diff
@@ -0,0 +1,109 @@
+
+// UITextView with slight top and bottom gradients covering scrolling text.
+class WMFGradientTextView : UITextView {
+    private let fadeHeight = 6.0
+    private var normalizedFadeHeight: Double {
+        return bounds.size.height > 0 ? fadeHeight /  Double(bounds.size.height) : 0
+    }
+
+    private lazy var gradientMask: CAGradientLayer = {
+        let mask = CAGradientLayer()
+        mask.startPoint = .zero
+        mask.endPoint = CGPoint(x: 0, y: 1)
+        mask.colors = [
+            UIColor.clear.cgColor,
+            UIColor.black.cgColor,
+            UIColor.black.cgColor,
+            UIColor.clear.cgColor
+        ]
+        layer.mask = mask
+        return mask
+    }()
+    
+    override func layoutSublayers(of layer: CALayer) {
+        super.layoutSublayers(of: layer)
+        guard layer == gradientMask.superlayer else {
+            assertionFailure("Unexpected superlayer")
+            return
+        }
+
+        // Keep fade heights fixed to `fadeHeight` regardless of text view height
+        gradientMask.locations = [
+            0.0,
+            NSNumber(value: normalizedFadeHeight),          // upper stop
+            NSNumber(value: 1.0 - normalizedFadeHeight),    // lower stop
+            1.0
+        ]
+
+        // Maintain fixed mask location on scroll
+        CATransaction.begin()
+        CATransaction.setDisableActions(true)
+        gradientMask.frame = CGRect(x: 0, y: contentOffset.y, width: bounds.size.width, height: bounds.size.height)
```

Comment:
It might be better to use the approach used in `RoundedCornerView` and update only when the size of the view changes without a `CATransaction`

Reply:
checking...

Reply:
The size is usually constant on scroll. The rounded corner view example seems like it wouldn't quite work even if switched to check position. The transaction keeps the mask re-positioning animation-free (https://stackoverflow.com/a/23291097/135557).

Hmm...

Reply:
I see now - I misunderstood and thought this was just adjusting for size and not content offset. Instead of repositioning the mask on scroll, could it be a view with a static mask that has this UITextView as a subview?

Reply:
Yes it could. I considered it but was trying to do it w/o extra view. The CATransaction doesn't seem to impact performance (though I haven't checked in instruments), but I'm totally fine with switching to separate view.

Reply:
fixed! i like it better now too :)

Reply:
no more having to fiddle with the scrolly bits!